Layer/Pdo/Iterator.php

```diff
@@ -0,0 +1,179 @@
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS AND CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace Hoa\Database\Layer\Pdo;
+
+use Hoa\Database;
+
+/**
+ * Class \Hoa\Database\Layer\Pdo\Iterator.
+ *
+ * Iterator Statement.
+ *
+ * @copyright  Copyright © 2007-2015 Hoa community
+ * @license    New BSD License
+ */
+class Iterator implements Database\IDal\WrapperIterator
+{
+    /**
+     * The statement instance.
+     *
+     * @var \PDOStatement
+     */
+    protected $_statement   = null;
+
+    /**
+     * The cursor orientation.
+     *
+     * @var int
+     */
+    protected $_orientation = 0;
+
+    /**
+     * The start cursor offset.
+     *
+     * @var int
+     */
+    protected $_offset      = 0;
+
+    /**
+     * The cursor row.
+     *
+     * @var array
+     */
+    protected $_row         = null;
+
+
+
+    /**
+     * Create an iterator instance.
+     *
+     * @param   \PDOStatement  $statement      The PDOStatement instance.
+     * @param   int            $orientation    This value must be one of the
+     *                                         DalStatement::FETCH_ORI_*
+     *                                         constants.
+     * @param   int            $offset         This value can be one of the
+     *                                         DalStatement::FETCH_OFFSET_*
+     *                                         constants or an arbitrary offset.
+     * @return  void
+     */
+    public function __construct(\PDOStatement $statement, $orientation, $offset)
+    {
+        $this->_statement   = $statement;
+        $this->_orientation = $orientation;
+        $this->_offset      = $offset;
+
+        return;
+    }
+
+    /**
+     * Get the statement instance.
+     *
+     * @return  \PDOStatement
+     */
+    protected function getStatement()
+    {
+        return $this->_statement;
+    }
+
+    /**
+     * Return the current element.
+     *
+     * @return  array
+     */
+    public function current()
+    {
+        return $this->_row;
+    }
+
+    /**
+     * Move forward to next element.
+     *
+     * @return  void
+     */
+    public function next()
+    {
+        $this->_row = $this->getStatement()->fetch(
+            \PDO::FETCH_ASSOC,
+            $this->_orientation
+        );
+
+        return;
+    }
+
+    /**
+     * Return always null.
+     *
+     * @return  null
+     */
+    public function key()
+    {
+        return null;
```

Comment:
Is an integer make sense? Not sure, because if we iterate backward, we don't really know what to do… So, ok for `null`, just wondering if you thought about this.